src/libsystemd-network/dhcp6-option.c

```diff
@@ -464,279 +452,291 @@ static int dhcp6_option_parse_ia_options(sd_dhcp6_client *client, const uint8_t
+                        if (r == -ENOMEM)
                                 return r;
-                        if (r >= 0 && lt_valid < lt_min)
-                                lt_min = lt_valid;
+                        if (r < 0)
+                                /* Ignore the sub-option on non-critical errors. */
+                                continue;
 
+                        lt_min = MIN(lt_min, a->iapdprefix.lifetime_valid);
+                        LIST_PREPEND(addresses, ia.addresses, a);
                         break;
-
+                }
                 case SD_DHCP6_OPTION_STATUS_CODE: {
                         _cleanup_free_ char *msg = NULL;
 
-                        r = dhcp6_option_parse_status(option->data, optlen, &msg);
-                        if (r < 0)
+                        r = dhcp6_option_parse_status(subdata, subdata_len, &msg);
+                        if (r == -ENOMEM)
                                 return r;
-                        if (r > 0) {
-                                if (ret_status_code)
-                                        *ret_status_code = r;
-
-                                log_dhcp6_client(client,
-                                                 "Received an IA option with non-zero status: %s%s%s",
-                                                 strempty(msg), isempty(msg) ? "" : ": ",
-                                                 dhcp6_message_status_to_string(r));
-
-                                return 0;
-                        }
+                        if (r < 0)
+                                log_dhcp6_client_errno(client, r,
+                                                       "Received an IA option with an invalid status sub option, ignoring: %m");
+                        else if (r > 0)
+                                return log_dhcp6_client_errno(client, SYNTHETIC_ERRNO(EINVAL),
+                                                              "Received an IA option with non-zero status: %s%s%s",
+                                                              strempty(msg), isempty(msg) ? "" : ": ",
+                                                              dhcp6_message_status_to_string(r));
                         break;
                 }
                 default:
-                        log_dhcp6_client(client, "Unknown IA option %d", opt);
-                        break;
+                        log_dhcp6_client(client, "Received an IA option with an unknown sub-option %u, ignoring", subopt);
                 }
+        }
+
+        if (IN_SET(option_code, SD_DHCP6_OPTION_IA_NA, SD_DHCP6_OPTION_IA_PD) &&
+            lt_t1 == 0 && lt_t2 == 0 && lt_min != UINT32_MAX) {
+                lt_t1 = lt_min / 2;
+                lt_t2 = lt_min / 10 * 8;
 
-                i += sizeof(*option) + optlen;
+                log_dhcp6_client(client, "Received an IA option with both T1 and T2 are zero. "
```

Comment:
"with both T1 and T2 equal to zero."
"Adjusting them to the valid lifetime…"

Reply:
WDYT about "Adjusting them based on the minimum valid lifetime of ..."?
For me, "Adjusting them to ..." sounds using the value as is.

Reply:
sounds good to me